platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,290 @@
+setting the `candidateRosterHash` field in the PlatformState.
+The `activeRosterHash` will be private to the platform and will not be settable or modifiable from outside the platform.
+
+This indirection avoids moving data around in the merkle state which requires data to be copied into the block stream,
+which is computationally expensive.
+Another benefit of this approach is that adoption trigger becomes very simple (app sets the roster) with delineated
+responsibilities between the app and the platform.
+
+Some edge cases worth considering include:
+
+1. multiple concurrent roster submission - has a potential for introducing race conditions. We will prevent this by
+   guaranteeing immutability on rosters.
+2. Size control on the map of rosters - we certainly don’t want this map to grow infinitely so insertion of rosters will
+   be controlled, with clear rules for removal of unused rosters (e.g. the acceptance of a new Candidate Roster
+   invalidates and removes existing one).
+
+###### Roster API - Implementation
+
+Implementation of some new TSS components has been proposed.
+See [link](https://www.notion.so/TSS-Platform-Architecture-04b15df371ba4b1d848360542e05a030?pvs=21).
+
+It is **important** to note that before the TSS protocol is implemented and launched, the Platform will adopt a
+Candidate Roster during any software upgrade if the Candidate Roster is present in the state. Only after the TSS
+protocol is launched will we be able to add an extra condition to verify that the Candidate Roster is complete, meaning
+that it has enough TSS key material and votes accumulated.
+
+### Core Behaviors, in summary
+
+- Roster Creation: App will create a Candidate Roster from the Candidate Address Book (CAB) and set it in the state.
+- Roster Submission: App will trigger roster adoption by setting the `candidateRosterHash` field in the PlatformState.
+- Roster Adoption: The platform will adopt the last submitted Candidate Roster when it is ready, based on the TSS key
+  material and vote accumulation.
+- Roster Replacement: If a new CAB is submitted before the previous one is adopted, the corresponding new Candidate
+  Roster will replace the old one.
+
+### Public API
+
+A new method will be added to the platform API to allow the App submit a Candidate Roster to the platform:
+
+```java
+//in State
+
+void submitCandidateRoster(@NonNull final Roster candidateRoster);
+```
+
+### Configuration
+
+The `config.txt` file will no longer be used for storing the address book in existing networks. It will only be used for
+the genesis of new networks.
+
+### Data storage
+
+All the data related to Active and Candidate Roster, including the key material being accumulated via TSS messages, will
+always be stored in the State.
+There will not be any other separate artifacts stored elsewhere (e.g., directly on disk.).
+The only artifact that will continue to be stored on disk separately from the State is the `config.txt`, which, as
```

Comment:
This is not the only artifact.  There are also the public and private RSA key files.   You should explain how this change also effects how the public and private keys are managed, if changed at all.   (Public key lifecycle is changed.)